alicloud/resource_alicloud_sag_acl.go

```diff
@@ -0,0 +1,108 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/smartag"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudSagAcl() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudSagAclCreate,
+		Read:   resourceAlicloudSagAclRead,
+		Update: resourceAlicloudSagAclUpdate,
+		Delete: resourceAlicloudSagAclDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateInstanceName,
+			},
+			"sag_count": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudSagAclCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	request := smartag.CreateCreateACLRequest()
+
+	request.Name = d.Get("name").(string)
+
+	raw, err := client.WithSagClient(func(sagClient *smartag.Client) (interface{}, error) {
+		return sagClient.CreateACL(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_sag_acl", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	response, _ := raw.(*smartag.CreateACLResponse)
+	d.SetId(response.AclId)
+	return resourceAlicloudSagAclRead(d, meta)
+}
+
+func resourceAlicloudSagAclRead(d *schema.ResourceData, meta interface{}) error {
+	sagService := SagService{meta.(*connectivity.AliyunClient)}
+	object, err := sagService.DescribeSagAcl(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	d.Set("name", object.Name)
+	d.Set("sag_count", object.SagCount)
+	return nil
+}
+
+func resourceAlicloudSagAclUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	update := false
+	request := smartag.CreateModifyACLRequest()
+	request.AclId = d.Id()
+
+	if d.HasChange("name") {
+		request.Name = d.Get("name").(string)
+		update = true
+	}
+	if update {
+		raw, err := client.WithSagClient(func(sagClient *smartag.Client) (interface{}, error) {
```

Comment:
I think the line 70-78 can be merged into one block `if d.HasChange("name") {...}`